### PR TITLE
airbyte-ci: fix integration test log buffering

### PR DESCRIPTION
**Pull request by postamar**
This PR fixes the log buffering issue in airbyte-ci that has plagued our recent source-mysql integration tests debugging efforts. What would happen is that we'd never see the `./gradlew ...:integrationTest` output in a timely manner, unlike all other pipeline steps.

As it turns out, the trigger for this buffering is that this container has secrets that were mounted to it. These secrets are the credentials downloaded from GSM so they are legit secrets. There's no way around this buffering, see commentary I added in the PR.

In any case, this PR adds a workaround when developing locally: we don't mount those files as secrets, just regular files. This is fine locally because those secrets are readily available on a development machine anyway. When airbyte-ci is not running locally, it's running in a github action or something and the buffering is something we can put up with in that case. In fact, so far, this hasn't been an issue at all in that context.

This PR includes another commit which has a trivial gradle fix. Its main purpose is to exercise this change on the snowflake connector. I'll do a couple of manual runs on python connectors just to make sure I didn't break anything there.

### CODE REVIEW

File: airbyte-ci/connectors/pipelines/pipelines/actions/environments.py [hunk 1]
```diff
@@ -1009,11 +1009,47 @@ def with_crane(
...
+    # however this becomes a nuisance locally: the developer wants the logs to be displayed to them in an as timely
+    # manner as possible. Since the secrets aren't really secret in that case anyway, we mount them in the container as
+    # regular files instead.
+    #
+    # The buffering behavior that comes into play when logs are scrubbed is both unavoidable and not configurable.
+    # It's fundamentally unavoidable because dagger needs to match a bunch of regexes (one per secret) and therefore
+    # needs to buffer at least as many bytes as the longest of all possible matches. Still, this isn't that long in
+    # practice in our case. The real problem is that the buffering is not configurable: dagger relies on a golang
+    # library called transform [1] to perform the regexp matching on a stream and this library hard-codes a buffer
+    # size of 4096 bytes for each regex [2].
```
Comment (alafanechere) [inline airbyte-ci/connectors/pipelines/pipelines/actions/environments.py:1026]: 🤯

File: airbyte-ci/connectors/pipelines/pipelines/actions/environments.py [hunk 2]
```diff
@@ -1009,11 +1009,47 @@ def with_crane(
...
+
+        contents = {}
+        for secret_file_name, secret in context.connector_secrets.items():
+            contents[secret_file_name] = await secret.plaintext()
+
+        def with_secrets_mounted_as_regular_files(container: Container) -> Container:
+            container = container.with_exec(["mkdir", secret_directory_path], skip_entrypoint=True)
+            for secret_file_name, secret in contents.items():
+                container = container.with_new_file(f"{secret_directory_path}/{secret_file_name}", secret, permissions=0o600)
+            return container
```
Comment (alafanechere) [inline airbyte-ci/connectors/pipelines/pipelines/actions/environments.py:1045]: nit: when I read this first I thought you mounted secrets from the local fs. This is not what's happening (and this is great): we're still pulling secrets from GSM at runtime and build a dict of `secret_file_name, secrets`. I'm just suggesting to add a comment here that remind this process.

File: airbyte-integrations/connectors/destination-snowflake/build.gradle [hunk 3]
```diff
@@ -24,6 +24,11 @@ application {
 
 }
 
+integrationTestJava {
+    // This is needed to make the destination-snowflake tests succeed - https://github.com/snowflakedb/snowflake-jdbc/issues/589#issuecomment-983944767
+    jvmArgs = ["--add-opens=java.base/java.nio=ALL-UNNAMED"]
+}
+
```
Comment (alafanechere) [inline airbyte-integrations/connectors/destination-snowflake/build.gradle:31]: Is it related to the secret logic change?

**Review by alafanechere — APPROVED** [review]
Congrats for finding the root cause!

File: airbyte-ci/connectors/pipelines/pipelines/actions/environments.py (hunk 2 above)
Comment (postamar) [reply airbyte-ci/connectors/pipelines/pipelines/actions/environments.py:1046]: Good point, will do.

File: airbyte-ci/connectors/pipelines/pipelines/actions/environments.py (hunk 1 above)
Comment (postamar) [reply airbyte-ci/connectors/pipelines/pipelines/actions/environments.py:1026]: I know, right... Go is full of this kind of mickey-mouse crap. 

FYI see also https://github.com/airbytehq/airbyte/issues/30394 which tracks the remaining work around this issue, essentially to ask the dagger maintainers to do something.

File: airbyte-integrations/connectors/destination-snowflake/build.gradle (hunk 3 above)
Comment (postamar) [reply airbyte-integrations/connectors/destination-snowflake/build.gradle:31]: No, this is just to trigger a CI run for snowflake.

File: buildSrc/src/main/groovy/airbyte-integration-test-java.gradle [hunk 4]
```diff
@@ -41,14 +41,20 @@ class AirbyteIntegrationTestJavaPlugin implements Plugin<Project> {
...
+                int numberThreads = 0
+                String numberThreadsString = project.property('numberThreads').toString()
+                if (numberThreadsString.isInteger()) {
+                    numberThreads = numberThreadsString as int
+                }
+                 if (numberThreads > 0 && numberThreads < maxParallelForks) {
+                    maxParallelForks = numberThreads
+                }
+            }
             maxHeapSize = '3g'
```
[the remark this answers is not in the document]
Comment (postamar) [reply buildSrc/src/main/groovy/airbyte-integration-test-java.gradle:57]: see above.

**Comment by postamar** [comment]
Green for source-google-sheets as well: https://github.com/airbytehq/airbyte/actions/runs/6187637825/job/16797910130 
I also ran locally to make sure I didn't break things there.

**Comment by postamar** [comment]
Thanks for the review! I've addressed all comments and ran some extra tests.

**Comment by octavia-approvington** [comment]
You did it! <br/> finish line

**Review by octavia-approvington — APPROVED** [review]

**Pull request merged**